### PR TITLE
get case base name from caseName with extension

### PR DESCRIPTION
**Pull request by basseche**
**Please check if the PR fulfills these requirements**

- [ ] The commit message follows our guidelines
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

**Does this PR already have an issue describing the problem?**

**What kind of change does this PR introduce?**

**What is the current behavior?**

**What is the new behavior (if this is a feature change)?**

**Does this PR introduce a breaking change or deprecate an API?**
- [ ] Yes
- [x] No

**If yes, please check if the following requirements are fulfilled**

- [ ] The *Breaking Change* or *Deprecated* label has been added
- [ ] The migration steps are described in the following section

**What changes might users need to make in their application due to this PR? (migration steps)**

**Other information**:

### CODE REVIEW

File: src/main/java/com/powsybl/caseserver/CaseController.java [hunk 1]
```diff
@@ -224,4 +225,13 @@ public ResponseEntity<List<CaseInfos>> getMetadata(@RequestParam("ids") List<UUI
         LOGGER.debug("get Cases metadata");
         return ResponseEntity.ok().body(caseService.getMetadata(ids));
     }
+
+    @GetMapping(value = "/cases/caseBaseName")
+    @Operation(summary = "Get case base name")
+    @ApiResponses(value = { @ApiResponse(responseCode = "200", description = "case base name"), })
```
Comment (ghazwarhili) [inline src/main/java/com/powsybl/caseserver/CaseController.java:231]: use @ApiResponse directly and enhance the description
`@ApiResponse(responseCode = "200", description ="Case base name retrieved successfully")`

File: src/main/java/com/powsybl/caseserver/CaseController.java [hunk 2]
```diff
@@ -224,4 +225,13 @@ public ResponseEntity<List<CaseInfos>> getMetadata(@RequestParam("ids") List<UUI
...
+    public ResponseEntity<String> getCaseBaseName(@RequestParam("caseName") String caseName) {
+        LOGGER.debug("getCaseBaseName request received");
```
Comment (ghazwarhili) [inline src/main/java/com/powsybl/caseserver/CaseController.java:233]: `    LOGGER.debug("Fetch base name for case: {}", caseName);`

File: src/main/java/com/powsybl/caseserver/CaseController.java [hunk 3]
```diff
@@ -224,4 +225,13 @@ public ResponseEntity<List<CaseInfos>> getMetadata(@RequestParam("ids") List<UUI
         LOGGER.debug("get Cases metadata");
         return ResponseEntity.ok().body(caseService.getMetadata(ids));
     }
+
+    @GetMapping(value = "/cases/caseBaseName")
```
Comment (ghazwarhili) [inline src/main/java/com/powsybl/caseserver/CaseController.java:229]: the value attribute in @GetMapping is redundant when there’s only one parameter
remove it
```suggestion
    @GetMapping("/cases/caseBaseName")
```

File: src/main/java/com/powsybl/caseserver/CaseController.java [hunk 4]
```diff
@@ -224,4 +225,13 @@ public ResponseEntity<List<CaseInfos>> getMetadata(@RequestParam("ids") List<UUI
...
+    @Operation(summary = "Get case base name")
+    @ApiResponses(value = { @ApiResponse(responseCode = "200", description = "case base name"), })
+    public ResponseEntity<String> getCaseBaseName(@RequestParam("caseName") String caseName) {
+        LOGGER.debug("getCaseBaseName request received");
+        String baseName = DataSourceUtil.getBaseName(caseName);
+        return ResponseEntity.ok().body(baseName);
```
Comment (ghazwarhili) [inline src/main/java/com/powsybl/caseserver/CaseController.java:235]: remove redundant body method call, it is unnecessary when returning the object directly
```suggestion
        return ResponseEntity.ok(baseName);
```

File: src/main/java/com/powsybl/caseserver/CaseController.java (hunk 2 above)
Comment (basseche) [reply src/main/java/com/powsybl/caseserver/CaseController.java:233]: Logger only take one String parameter !
  And i am not sûre we are fetching something here ! it is server side

Comment (basseche) [reply src/main/java/com/powsybl/caseserver/CaseController.java:233]: but it is server side we are not fetching anything here !

File: src/main/java/com/powsybl/caseserver/CaseController.java (hunk 1 above)
Comment (basseche) [reply src/main/java/com/powsybl/caseserver/CaseController.java:231]: Ok but i don't like the term successfully i will put only retrieved

File: src/main/java/com/powsybl/caseserver/CaseController.java (hunk 2 above)
Comment (ghazwarhili) [reply src/main/java/com/powsybl/caseserver/CaseController.java:233]: It was a suggestion to add LOGGER.debug, and you can choose the appropriate message.

**Review by ghazwarhili — APPROVED** [review]

**Pull request merged**